packages/dataviews/src/hooks/use-is-form-valid.ts

```diff
@@ -0,0 +1,240 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { normalizeFields } from '../normalize-fields';
+import type { Field, Form, FormValidity } from '../types';
+
+/**
+ * Type for the return value of useIsFormValid hook.
+ */
+
+/**
+ * Hook that validates a form item and returns an object with error messages for each field.
+ *
+ * @param item   The item to validate.
+ * @param fields Fields config.
+ * @param form   Form config.
+ *
+ * @return Record of field IDs to error messages (undefined means no error).
+ */
+export function useIsFormValid< Item >(
```

Comment:
I think the name should be changed to `useValidation`, it doesn't return just a boolean. or `useValidity`. Consistency with the prop is probably better but I prefer "validation" over "validity" which sounds like a weird word.